packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -240,140 +242,48 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
-             */
-            const names = match[6]?.trim()
-            dynamicImports[
-              dynamicImportTreeshakenRE.lastIndex - match[5]?.length
-            ] = { declaration: `const {${names}}`, names: `{ ${names} }` }
-          }
-        }
+        for (const imp of imports) {
+          const { s: start, e: end, se: expEnd } = imp
 
-        let s: MagicString | undefined
-        const str = () => s || (s = new MagicString(source))
-        let needPreloadHelper = false
-
-        for (let index = 0; index < imports.length; index++) {
-          const {
-            s: start,
-            e: end,
-            ss: expStart,
-            se: expEnd,
-            d: dynamicIndex,
-            a: attributeIndex,
-          } = imports[index]
-
-          const isDynamicImport = dynamicIndex > -1
-
-          // strip import attributes as we can process them ourselves
-          if (!isDynamicImport && attributeIndex > -1) {
-            str().remove(end + 1, expEnd)
+          const isDynamicImport = imp.d > -1
+          const hasAttributes = imp.a > -1
+
+          // strip import attributes as we process them ourselves
+          if (!isDynamicImport && hasAttributes) {
+            s.remove(end + 1, expEnd)
           }
 
           if (
             isDynamicImport &&
-            insertPreload &&
+            willInsertPreload &&
             // Only preload static urls
             (source[start] === '"' ||
               source[start] === "'" ||
               source[start] === '`')
           ) {
-            needPreloadHelper = true
-            const { declaration, names } = dynamicImports[expEnd] || {}
-            if (names) {
-              /* transform `const {foo} = await import('foo')`
-               * to `const {foo} = await __vitePreload(async () => { const {foo} = await import('foo');return {foo}}, ...)`
-               *
-               * transform `import('foo').then(({foo})=>{})`
-               * to `__vitePreload(async () => { const {foo} = await import('foo');return { foo }},...).then(({foo})=>{})`
-               *
-               * transform `(await import('foo')).foo`
-               * to `__vitePreload(async () => { const {foo} = (await import('foo')).foo; return { foo }},...)).foo`
-               */
-              str().prependLeft(
-                expStart,
-                `${preloadMethod}(async () => { ${declaration} = await `,
-              )
-              str().appendRight(expEnd, `;return ${names}}`)
-            } else {
-              str().prependLeft(expStart, `${preloadMethod}(() => `)
+            if (!importedPreloadHelper) {
+              s.prepend(`
+                import { ${preloadMethod} } from "${preloadHelperId}";
+  
+                // To prevent dead code elimination. Will properly remove in generateBundle
+                // It's referenced twice so it doesn't get inlined by the minifier
+                ${preloadSaver}(${preloadMethod},${preloadMethod});
```

Comment:
While this can make the `preloadMethod` import to be kept, since this function call is treated as a sideeffect, rollup will treat this module as sideeffectful and would make the chunking less ideal.